api-authorizations/src/main/scala/com/normation/plugins/apiauthorizations/UserTokenApiDefinition.scala

```diff
@@ -128,67 +111,172 @@ class UserApi(
+  object RestApiAccount extends UserJsonCodec {
+    implicit class ApiAccountOps(val account: ApiAccount) extends AnyVal {
+      import ApiAccountKind._
+      import io.scalaland.chimney.syntax._
+      def expirationDate: Option[String] = {
+        account.kind match {
+          case PublicApi(_, expirationDate) => expirationDate.map(DateFormaterService.getDisplayDateTimePicker)
+          case User | System                => None
+        }
+      }
 
-        case eb: EmptyBox =>
-          val e = eb ?~! s"Error when trying to get user '${authzToken.actor.name}' API token"
-          RestUtils.toJsonError(None, e.messageChain)(schema.name, true)
+      def expirationDateDefined: Boolean = expirationDate.isDefined
+
+      def authzType: Option[ApiAuthorizationKind] = {
+        account.kind match {
+          case PublicApi(authz, _) => Some(authz.kind)
+          case User | System       => None
+        }
+      }
+
+      def acl: Option[List[JsonApiAcl]] = {
+        import ApiAuthorization._
+        account.kind match {
+          case PublicApi(authz, expirationDate) =>
+            authz match {
+              case None | RO | RW => Option.empty
+              case ACL(acls)      => Some(acls.flatMap(x => x.actions.map(a => JsonApiAcl(x.path.value, a.name))))
+            }
+          case User | System                    => Option.empty
+        }
       }
+
+      /**
+        * Always hides any hashed token, and displays any clear-text token
+        */
+      def toRest = account.transformInto[RestApiAccount]
+
+      /**
+        * Always displays the passed secret token
+        */
+      def toRestWithSecret(secret: String) = account.copy(token = ApiToken(secret)).transformInto[RestApiAccount]
     }
+
+    implicit val transformer: Transformer[ApiAccount, RestApiAccount] = Transformer
+      .define[ApiAccount, RestApiAccount]
+      .withFieldComputed(_.kind, _.kind.kind)
+      .withFieldComputed(_.acl, _.acl)
+      .withFieldComputed(_.expirationDate, _.expirationDate)
+      .withFieldComputed(_.expirationDateDefined, _.expirationDateDefined)
+      .withFieldComputed(
+        _.authorizationType,
+        _.authzType
+      )
+      .buildTransformer
+
+    implicit val publicTokenEncoder: JsonEncoder[ApiToken] =
+      JsonEncoder[String].contramap(_.value)
+
+    implicit val encoder: JsonEncoder[RestApiAccount] = DeriveJsonEncoder.gen[RestApiAccount]
+  }
+
+  final case class RestAccountsResponse private (
+      accounts: List[RestApiAccount]
+  )
+  class RestAccountsResRestApiAccount(
+      accounts: List[RestApiAccount]
+  )
+
+  object RestAccountsResponse {
+    import RestApiAccount._
+
+    implicit val encoder: JsonEncoder[RestAccountsResponse] = DeriveJsonEncoder.gen[RestAccountsResponse]
+
+    private def apply(accounts: List[ApiAccount], secret: Option[String] = None): RestAccountsResponse = new RestAccountsResponse(
+      accounts.map(a => secret.map(a.toRestWithSecret(_)).getOrElse(a.toRest))
```

Comment:
Can you add a comment explaining that the list is needed b/c the answer expects a json array ?